eclipse/plugin/src/nl/tudelft/watchdog/eclipse/logic/ui/listeners/staticanalysis/ResourceAndResourceDeltaVisitor.java

```diff
@@ -0,0 +1,125 @@
+package nl.tudelft.watchdog.eclipse.logic.ui.listeners.staticanalysis;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IMarker;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.IResourceDelta;
+import org.eclipse.core.resources.IResourceDeltaVisitor;
+import org.eclipse.core.resources.IResourceVisitor;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IPath;
+import org.joda.time.DateTime;
+import nl.tudelft.watchdog.core.logic.document.Document;
+import nl.tudelft.watchdog.core.logic.event.TrackingEventManager;
+import nl.tudelft.watchdog.core.logic.event.eventtypes.staticanalysis.FileWarningSnapshotEvent;
+import nl.tudelft.watchdog.core.logic.ui.listeners.staticanalysis.CoreMarkupModelListener;
+import nl.tudelft.watchdog.core.logic.ui.listeners.staticanalysis.StaticAnalysisMessageClassifier;
+import nl.tudelft.watchdog.core.logic.ui.listeners.staticanalysis.Warning;
+import nl.tudelft.watchdog.eclipse.logic.document.DocumentCreator;
+
+public class ResourceAndResourceDeltaVisitor implements IResourceDeltaVisitor, IResourceVisitor {
+	
+	private final TrackingEventManager trackingEventManager;
+	private Map<IPath, List<MarkerHolder>> currentFileMarkers;
+	private final boolean shouldCreateSnapshot;
+
+    public ResourceAndResourceDeltaVisitor(TrackingEventManager trackingEventManager,
+    		Map<IPath, List<MarkerHolder>> currentFileMarkers, boolean shouldCreateSnapshot) {
+		this.trackingEventManager = trackingEventManager;
+		this.currentFileMarkers = currentFileMarkers;
+		this.shouldCreateSnapshot = shouldCreateSnapshot;
+    }
+
+    /**
+     * Can be new resources, deleted or simply changed. In any case, even if we do not have
```

Comment:
What can be a new resource, ...? The trigger of this method? If so, please document